File: 1342 - Numbers of Steps to Reduce a Number to Zero/NumberOfSteps.js
// https://leetcode.com/problems/number-of-steps-to-reduce-a-number-to-zero/

function numberOfSteps1(num) {
  let steps = 0;

  while (num > 0) {
    const isEven = num % 2 === 0;

    if (isEven) {
      num /= 2;
    } else {
      num--;
    }

    steps++;
  }

  return steps;
}

function numberOfSteps2(num) {
  let steps = 0;

  while (num > 0) {
    const isEven = (num & 1) === 0;

    if (isEven) {
      num >>= 1;
    } else {
      num--;
    }

    steps++;
  }

  return steps;
}

console.log(numberOfSteps1(30));

File: 1342 - Numbers of Steps to Reduce a Number to Zero/NumberOfSteps.java
public class NumberOfSteps {
    public static void main(String[] args) {
        System.out.println(numberOfSteps2(30));
    }

    public static int numberOfSteps1(int num) {
        int steps = 0;

        while (num > 0) {
            boolean isEven = num % 2 == 0;

            if (isEven) {
                num /= 2;
            } else {
                num--;
            }

            steps++;
        }

        return steps;
    }

    public static int numberOfSteps2(int num) {
        int steps = 0;

        while (num > 0) {
            boolean isEven = (num & 1) == 0;

            if (isEven) {
                num >>= 1;
            } else {
                num--;
            }

            steps++;
        }

        return steps;
    }
}